tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java

```diff
@@ -27,21 +44,80 @@ public void run() {
         try (final var inputStream = connection.getInputStream();
-             final var outputStream = connection.getOutputStream()) {
+             final var outputStream = connection.getOutputStream();
+             final BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))
+        ) {
+            final HttpRequest request = HttpRequest.from(bufferedReader);
 
-            final var responseBody = "Hello world!";
-
-            final var response = String.join("\r\n",
-                    "HTTP/1.1 200 OK ",
-                    "Content-Type: text/html;charset=utf-8 ",
-                    "Content-Length: " + responseBody.getBytes().length + " ",
-                    "",
-                    responseBody);
+            final HttpResponse response = handleRequest(request);
 
             outputStream.write(response.getBytes());
             outputStream.flush();
-        } catch (IOException | UncheckedServletException e) {
+        } catch (IOException | UncheckedServletException | URISyntaxException e) {
             log.error(e.getMessage(), e);
         }
     }
+
+    public HttpResponse handleRequest(final HttpRequest request) throws URISyntaxException {
+
+        final String uriPath = request.getPath();
+
+        if (uriPath.equals("/login")) {
+            final String path = LOGIN_PAGE;
+            if (request.getMethod().equals("GET")) {
+                return HttpResponse.of(HttpStatus.OK, path);
+            }
+            if (request.getMethod().equals("POST")) {
+                final boolean isAuthenticated = processLogin(request);
+                final String redirectUrlPath = isAuthenticated ? INDEX_PAGE : UNAUTHORIZED_PAGE;
+                return HttpResponse.of(HttpStatus.FOUND, redirectUrlPath);
+            }
+        }
+
+        if (uriPath.equals("/register")) {
+            final String path = REGISTER_PAGE;
+            if (request.getMethod().equals("GET")) {
+                return HttpResponse.of(HttpStatus.OK, path);
+            }
+            if (request.getMethod().equals("POST")) {
+                processRegister(request);
+                return HttpResponse.of(HttpStatus.FOUND, INDEX_PAGE);
+            }
+        }
+
+        return HttpResponse.of(HttpStatus.OK, uriPath);
+    }
+
+    public boolean processLogin(final HttpRequest request) {
+        if (!request.containsBody(ACCOUNT_KEY) || !request.containsBody(PASSWORD_KEY)) {
+            return false;
+        }
+        final String account = request.getBody(ACCOUNT_KEY);
+        final String password = request.getBody(PASSWORD_KEY);
+        Optional<User> userOptional = InMemoryUserRepository.findByAccount(account);
+
+        if (userOptional.isPresent()) {
+            User user = userOptional.get();
+            if (user.checkPassword(password)) {
+                log.info(user.toString());
+                return true;
+            }
+        }
+        return false;
```

Comment:
Optional의 메서드를 적절하게 사용해서 아래와 같이 간소화 시켜보는건 어떨까요?

```suggestion
        return InMemoryUserRepository.findByAccount(account)
                .map(user -> user.checkPassword(password))
                .orElse(false);
```